### ISSUE TITLE
After accepting a request to join a circle, the notification remains active

### ISSUE THREAD

**Issue by jerdesign** [nextcloud/circles#810]
### Describe the bug
After accepting a request from somebody else to join a circle, the notification remains active. Accepting the request via the notification interface a second time results in an error message.

**To Reproduce**
1. Somebody sends you a request to join one of your circles
2. You accept the request to join the cirlce
3. Reload the page → the notification will still be there
4. Accept the request to join the cirlce again → leads to error message

**Expected behavior**
After accepting a request to join a circle the notification disappears.

**Actual behavior**
After accepting a request to join a circle the notification does not disappear. Clicking on »Accept« results in an error message.

**Screenshots**
![CleanShot 2021-09-12 at 20 18 21@2x_Edit](https://user-images.githubusercontent.com/4391042/133001339-cb87fda3-27fb-4997-a0b4-e8cd8ea4034b.png)

### Server configuration

**Operating system**: RHEL8

**Web server:** Nginx

**Database:** MariaDB 

**PHP version:** 7.4 fpm

**Nextcloud version:** 22.1.0

**Contacts version:** 4.0.0

**Updated since Nextcloud 20**

**List of activated apps:**
```
Enabled:
  - accessibility: 1.7.0
  - admin_audit: 1.11.0
  - circles: 22.1.0
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contacts: 4.0.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.18.0
  - encryption: 2.9.0
  - external: 3.9.0
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_pdfviewer: 2.3.0
  - files_rightclick: 1.1.0
  - files_sharing: 1.13.2
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.11.0
  - group_everyone: 0.1.8
  - logreader: 2.7.0
  - lookup_server_connector: 1.9.0
  - nextcloud_announcements: 1.11.0
  - notifications: 2.10.1
  - oauth2: 1.9.0
  - oidc_login: 2.0.4
  - provisioning_api: 1.11.0
  - recommendations: 1.1.0
  - serverinfo: 1.12.0
  - settings: 1.3.0
  - spreed: 12.0.1
  - systemtags: 1.11.0
  - text: 3.3.0
  - theming: 1.12.0
  - theming_customcss: 1.9.0
  - twofactor_backupcodes: 1.10.1
  - viewer: 1.6.0
  - welcome: 0.0.6
  - workflowengine: 2.3.0
Disabled:
  - activity
  - collectives
  - federation
  - files_external
  - firstrunwizard
  - password_policy
  - photos
  - privacy
  - sharebymail
  - support
  - survey_client
  - updatenotification
  - user_ldap
  - user_status
  - weather_status
```

**Nextcloud configuration:**
```
<?php
$CONFIG = array (
	'config_is_read_only' => true,

	'trusted_domains' =>
	array (
		0 => 'FUU',
		1 => 'BAR',
		2 => 'FUU',
	),
	'overwrite.cli.url' => 'FUU',
	'overwriteprotocol' => 'https',
	'overwritehost' => 'BAR',
	'trusted_proxies' => ['PROXY_IP'],

	'default_language' => 'de_DE',
	'default_locale' => 'de_DE',
	'default_phone_region' => 'DE',
	'skeletondirectory' => '',
	'knowledgebaseenabled' => false,

	'appstoreenabled' => false,

	'log_type' => 'errorlog',
	'logfile' => '',
	'loglevel' => 1,
	'debug' => false,

  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'info',
  'mail_domain' => 'DOMAIN',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'HOST',
  'mail_smtpport' => 'PORT',
  'mail_smtpname' => 'NAME',
  'mail_smtppassword' => 'PASSWORD',


	// OIDC Login
  'allow_user_to_change_display_name' => false,
  'lost_password_link' => 'disabled',
  'oidc_login_disable_registration' => false,
  'oidc_login_auto_redirect' => true,
  'oidc_login_redir_fallback' => false,
  'oidc_create_groups' => true,
  'oidc_login_hide_password_form' => true,
  'oidc_login_client_id' => 'CLIENT_ID',
  'oidc_login_client_secret' => 'SECRET',
  'oidc_login_provider_url' => 'URL',
  'oidc_login_logout_url' => 'URL',
  'config_is_read_only' => true,
  'oidc_login_attributes' => array(
    'id' => 'preferred_username',
    'name' => 'name',
    'mail' => 'email',
    'groups' => 'groups'
);
```

### Client configuration
**Browser:**
Brave 1.28.106 Chromium: 92.0.4515.159 (Offizieller Build) (arm64)

**Operating system:**
macOS Monterey 12.0 Beta

**Comment by joshtrichards** [comment]
Code is kinda there, but commented out. Not sure the state of it:

https://github.com/nextcloud/circles/blob/7cc0a359e162c17041f34cb17b8dd5bfe506e43b/lib/Service/NotificationService.php#L185-L194

Dev docs are here if anyone feels like making an attempt at fixing:

https://github.com/nextcloud/notifications/blob/master/docs/notification-workflow.md#marking-a-notification-as-readdeletedprocessedobsoleted

In the mean time, the X can be used a workaround on the notification (after clicking Accept once of course - since it otherwise does work... just the Accept doesn't automatically dismiss the notification)

**Comment by kesselb** [comment]
Thanks for your bug report :+1: 

Fixed for the upcoming 29.0.5 and 28.09 (both planned for 2024-08-20),
